### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by ReubenTeng**
Updated design - model, implementation - appointment as well as Glossary

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -387,57 +393,40 @@ Makes use of many methods from `UniqueList`, e.g. `add`, `setElement`, `remove`,
...
-The `AppointmentManager` class contains a summary of all the "back-end" logic of appointment commands on the app's `UniqueList` of `Appointment`s. This follows the SRP, as everything related to the execution of appointment commands can be found here. This also forces the customising of code to fit exactly the purposes needed for appointment commands, even if the methods simply call a `UniqueList` method that fulfills the exact purpose.
+* Pros: Keeps stored data perpetually up-to-date.
 
-#### 4.3.2. Current Implementation
+* Cons: Constant comparision become computationally intensive when there are many `UPCOMING` appointments. 
+  
+Option 2: Check `Datetime` of `Appointment` against current `Datetime` only when `getStatus()` method called
 
-Makes use of many methods from `UniqueList`, e.g. `add`, `setElement`, `remove`, `sort`.
+* Pros: Computation is performed when the value of `status` is needed, resulting in reduced performance cost.
```
Comment (kaitlynng) [inline docs/DeveloperGuide.md:418]: "increased performance" or "reduced computational cost"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -387,57 +393,40 @@ Makes use of many methods from `UniqueList`, e.g. `add`, `setElement`, `remove`,
...
 
-#### 4.3.3. Design Consideration
+* Cons: `status` of `Appointment` stored within the application may be outdated, depending on when getStatus() was called.
```
Comment (kaitlynng) [inline docs/DeveloperGuide.md]: There's no more "status" stored in Appointment - status is only computed at the point of getStatus() (there is no other time status can be accessed), so whenever status of an appointment is retrieved it is always updated.

Comment (kaitlynng) [reply docs/DeveloperGuide.md]: The con I guess would be that it is a less intuitive architecture as now, appointments don't actually have an inherent status - they just have a field that specifies whether the appointment is missed, and the status is computed when necessary.

**Review by kaitlynng — COMMENTED** [review]
Didn't check the AppointmentObjectDiagram

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ReubenTeng) [reply docs/DeveloperGuide.md]: Requested changes have been added!

**Review by kaitlynng — APPROVED** [review]

**Pull request merged**